pyccel/ast/core.py

```diff
@@ -2271,6 +2210,61 @@ def prefix(self):
         """
         return self._prefix
 
+class ConstructorCall(DottedFunctionCall):
+
+    """
+    It represents a constructor for undefined function classes.
+
+    This class is used to create instances of undefined function classes, providing the ability to
+    pass specific arguments to the constructor and associate a class variable. It is a subclass of
+    the DottedFunctionCall and inherits its properties and behavior.
+
+    Parameters
+    ----------
+    func : FunctionDef, str
+        An instance of FunctionDef or function name.
+
+    arguments : list, tuple, None
+        A list of arguments.
+
+    cls_variable : CustumDataType, optional
+        An instance of `CustumDataType` representing the class variable associated with the constructor.
+    """
+    __slots__ = ('_cls_variable')
+    _attribute_nodes = ()
+
+    is_commutative = True
```

Comment:
Do you know what this class variable is for and where/if it is used?

Reply:
The `_cls_variable` is used in the `pyccel/codegen/printing/pycode.py` but the data in `_cls_variable` are the same as the arguments of `ConstructorCall` class so the `_cls_variable` could be deleted. https://github.com/pyccel/pyccel/blob/0e71d4819472d8e35563e5b975bf4ac4a5df17c5/pyccel/parser/semantic.py#L2340-L2341

Reply:
My comment was about the variable `is_commutative`, but you are correct the `cls_variable` argument could return `_args[0]` to avoid having unnecessary variables and having this class appear twice in the "users" of `cls_variable`

Reply:
Fixed